lib/trivial.nix

```diff
@@ -232,6 +232,30 @@ in {
 
   nixpkgsVersion = builtins.trace "`lib.nixpkgsVersion` is deprecated, use `lib.version` instead!" version;
 
+  /* The output path of a derivation to fetch this nixpkgs if available,
+     otherwise the store path to this nixpkgs. This can be used to incur a
+     dependency on this nixpkgs.
+
+     This is different than pkgs.path in the case of flakes-based nixpkgs,
+     since pkgs.path is a path to nixpkgs without derivation context, and thus
+     its usage typically, undesirably, results in copying this nixpkgs.
+
+     See: https://github.com/NixOS/nix/issues/5868
+          https://github.com/NixOS/nix/issues/9428
+  */
+  selfPath = if lib.trivial.inPureEvalMode || !(lib.strings.isStorePath (toString ../.))
```

Comment:
Regarding the name:
- It's good that this value is in `lib` and not `pkgs`. There's no reason for `pkgs.path` to be in Nixpkgs.
  - We should consider moving `pkgs.path` to `lib` in the future, and in that case it should be obvious that it's closely related to `selfPath`
- `pkgs.path` is a path value, but `selfPath` is not a path value, but rather a string (that has a path in it). So it ending with `Path` is a bit misleading.
- The `self` word is kind of Flakes-specific, when it's also available without Flakes. In particular it can easily be confused with the overlay's self/super terminology.

How about `lib.trivial.nixpkgsStorePathString`? And then we could consider renaming `pkgs.path` to `lib.trivial.nixpkgsPath` in the future.

Reply:
Yeah, that sounds like an excellent plan, changed.